### ISSUE TITLE
Add ability to confirm/rollback grain upgrades

### ISSUE THREAD

**Issue by ndarilek** [sandstorm-io/sandstorm#1913]
So Rocket.chat just completely broke their UI for screen readers--as in, most of the links are entirely unusable.

I wonder if we might mitigate this by confirming grain upgrades? Basically, a grain runs with the new version of the spk, but the old data/spk is held in reserve. A "Confirm Upgrade" button appears and lets you remove the old data, while a Rollback button points the grain at its older spk/data with an Upgrade button displayed for when you're ready to switch back. I think it'd be sufficient to just retain a single version, though obviously there are some edge cases I haven't considered, like what to do if a grain isn't checked but is upgraded past several versions. Maybe the spk/data for a given grain is retained, then once no grains are using it the data gets GC'd? So you could keep multiple spk versions around as long as some data requires them?

It's not a huge deal I suppose, though I'd certainly be hosed now if this app was core to my business or something. :) With self-hosting I could restore a backup of both the app and my data, and download older app versions if needed, but I'm not sure how I'd do the same with Sandstorm if an app upgrades to an inaccessible version.

Thanks.

**Comment by paulproteus** [comment]
I just wanted to chime in to say: I agree that this is important. I'll see
how we can prioritize this.

**Comment by zeigerpuppy** [comment]
I agree, a very important feature.  There was another example of an update breaking functionality today (https://github.com/synchrone/sandstorm-radicale/issues/22).
It would be good to have the option for the administrator to rollback to an earlier release of an application (1 version back should be enough)
